[rocThrust] Download specific rocPRIM branch (#722)
ROCm/rocm-libraries: projects/rocthrust/CHANGELOG.md

Documentation for rocThrust available at
[https://rocm.docs.amd.com/projects/rocThrust/en/latest/](https://rocm.docs.amd.com/projects/rocThrust/en/latest/).

## rocThrust 4.1.0 for ROCm 7.1

### Added

> * rocThrust, as well as its dependencies rocPRIM and rocRAND have been moved into the new rocm-libraries "monorepo" repository (https://github.com/ROCm/rocm-libraries). This repository contains a number of ROCm libraries that are frequently used together.

Reviewer: ```suggestion
* rocThrust and its dependent libraries, rocPRIM and rocRAND, have moved to a new rocm-libraries "monorepo" repository: https://github.com/ROCm/rocm-libraries. This repository contains several ROCm libraries that are frequently used together.
```

>   * The repository migration requires a few changes to the way that rocThrust's ROCm library dependencies are fetched.

Reviewer: ```suggestion
  * Changes have been made to the way that rocThrust fetches library dependencies:
```
